test/Sentry.Tests/Internals/SentryStackTraceFactoryTests.cs

```diff
@@ -45,6 +45,28 @@ public void Create_NoExceptionAndAttachStackTraceOptionOnWithOriginalMode_Curren
             ) == true);
     }
 
+    [Theory]
+    [InlineData(StackTraceMode.Original, "Create_Async_CurrentStackTrace { <lambda> }")]
+    [InlineData(StackTraceMode.Enhanced, "void SentryStackTraceFactoryTests.Create_Async_CurrentStackTrace(StackTraceMode mode, string method)+() => { }")]
+    public void Create_Async_CurrentStackTrace(StackTraceMode mode, string method)
+    {
+        _fixture.SentryOptions.AttachStacktrace = true;
+        _fixture.SentryOptions.StackTraceMode = mode;
+        var sut = _fixture.GetSut();
+
+        SentryStackTrace stackTrace = null!;
+        Task.Run(() => stackTrace = sut.Create()).Wait();
```

Comment:
This isn't really async.  Perhaps you should change the test method from void to `async Task`, and do something like this instead?

```csharp
await Task.Run(async () =>
{
    await Task.Yield();
    return stackTrace = sut.Create();
});
```

Of course, that will change all your expected values.

Reply:
Would that still test the case you were trying to capture?

Reply:
I was able to update the regex to fix that.  Thanks.

Reply:
Actually, after discussing with Bruno, my change would affect grouping and thus can't be done.  Yours is fine though, as we haven't seen that before until Unity.